vendingMachine/vendingMachineV3/src/impl/AdminImpl.java

```diff
@@ -0,0 +1,38 @@
+package impl;
+
+
+import entity.Coin;
+import entity.Product;
+import interfaces.Admin;
+import interfaces.CoinManagement;
+import interfaces.ProductManagment;
+
+public class AdminImpl implements Admin {
```

Comment:
Here admin still can access other functionality like all methods from ProductManagement and CoinManagement, 
even though it's implementing an interface.

Reply:
I have to get both of your views on this. Let discuss this on-call.